src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -333,6 +333,10 @@ private static byte[] EncodeStringName()
         {
             var type = typeof(string);
             var name = type.AssemblyQualifiedName;
+            
+            // ResourceWriter always write the resources with legacy core library name (i.e. mscorlib).
+            // Ensure that when we read the type name, we update the library name to be the same.
+            name = name.Replace("System.Private.CoreLib", "mscorlib");
```

Comment:
I wanted to quickly try out a change (which is what this is). I think it should be:

var name = "System.String, mscorlib, Version="